literals.go

```diff
@@ -71,6 +71,12 @@ type TypedLiteral[T LiteralType] interface {
 	Comparator() Comparator[T]
 }
 
+type NumericLiteral interface {
```

Comment:
do we need this type?

Reply:
It's used with the transforms, specifically for `truncateNumber`.

Looking through this, I could probably change this to not be exported though if we want.